cuda/solver/cb_gmres_kernels.cu

```diff
@@ -480,11 +480,9 @@ void calculate_qy(ConstAccessor3d krylov_bases, size_type num_krylov_bases,
         before_preconditioner->get_stride();
 
     constexpr auto block_size = default_block_size;
-    const dim3 grid_dim{
-        static_cast<unsigned int>(
-            ceildiv(num_rows * stride_before_preconditioner, block_size)),
-        1, 1};
-    const dim3 block_dim{block_size, 1, 1};
+    const auto grid_dim = static_cast<unsigned int>(
+        ceildiv(num_rows * stride_before_preconditioner, block_size));
+    const auto block_dim = block_size;
 
 
     calculate_Qy_kernel<block_size><<<grid_dim, block_dim>>>(
```

Comment:
check?

Reply:
Much of that file will change when I finish #861, so I left it untouched for now